traversal/scanner/GraphIterator.java

```diff
@@ -206,28 +206,51 @@ private void initialiseStart() {
     private void initialiseEnds() {
         assert toTraverse.isEmpty();
         toRevisit.addAll(procedure.endVertices());
-        direction = Direction.REVISIT;
+        direction = Direction.REVISIT_RETRIEVED;
```

Comment:
`initialiseEnds` is called when a full answer is found and we look for the next complete answer